TFNetwork.py

```diff
@@ -985,6 +985,9 @@ def inv_reduce_sum(x, name):
           d["cost:%s" % loss_name] = reduce_sum(loss.get_loss_value_for_fetch(), name="cost:%s" % loss_name)
         if loss.get_error_value() is not None:
           d["error:%s" % loss_name] = reduce_sum(loss.get_error_value(), name="error:%s" % loss_name)
+        if config.bool("debug_display_loss_per_layer", False):
+          if loss.get_loss_value_for_objective() is not None:
+            d["loss:%s" % loss_name] = reduce_sum(loss.get_loss_value_for_objective(), name="loss:%s" % loss_name)
```

Comment:
The key name is confusing. It's already confusing that we have sort of "cost" == "loss", but this makes it more confusing.
Better name it `"cost_objective:..."` or so.
But anyway, is this really needed? I think adding it via `tf.summary.scalar` is probably better, right?
Note also that the difference is not necessarily that one is normalized and the other is not.

Reply:
I used this to be able to compare the actual loss of two different objectives and tune my loss scaling accordingly. The "cost" values do not include the loss scaling and seemed to be normalized with respect to the sequence length. I did not see another way to solve my issue.

Reply:
This does not address any of my points?

* The key name is confusing. It's already confusing that we have sort of "cost" == "loss", but this makes it more confusing.
Better name it "cost_objective:..." or so.
* Is `tf.summary.scalar` maybe better? (That would have the advantage that the code is much simpler.)
* Note also that the difference is not necessarily that one is normalized and the other is not.

Reply:
I refered to "But anyway, is this really needed?". Besides of how to use tf.summary.scalar (the tensorflow page did not help), I understand your points.

Reply:
Do you know TensorBoard? See TF documentation about that. Putting stuff on stdout is somewhat limited, and I think for more extra info, TensorBoard (via `tf.summary`) should always be used. That's what we used already a lot (e.g. stuff like gradient norms, or basically anything).

Reply:
Yes, I used Tensorboard once but it did not show more than the logs/learning-rate file itself for me, so I never used it again (are there flags for enabling more values?). But if you want it do be added only there it is ok, and I guess the option can be removed. The question is then how to distinguish the losses? "score" (in the learning rates file), "cost" (log) and "loss" (tensorboard) are all the same, so how should I call the actual loss? "objective_loss"? Also, they would be equal when `loss.use_normalized_loss` is `True`, but that should be ok.

Reply:
TensorBoard should always show all losses.
See all `tf.summary.scalar` usages.
You will find (among other things) this:
```
tf.summary.scalar("loss_%s" % name, self._loss_value * self._norm_factor)
```
If you have your debug option enabled, you could add this:
```
tf.summary.scalar("loss_unnormalized_%s" % name, self._loss_value)
```

Reply:
Yes, I already checked the Tensorboard again, and also the function `_tf_summary` of the `LossHolder`. My question was how you can enable more than the all losses that are already part of the logs. (e.g, the gradient norms you mentioned)

Also unnormalized might be the wrong term, as (if I understand correctly) the actual loss can be normalized. Or do you want to display normalized and un-normalized, and let the user have to know which is used for gradient computation and which is not?

Reply:
Just see all usages of `tf.summary.scalar`. E.g. there is `debug_grad_summaries`.
`_loss_value` at that point is not normalized. In any case, the only difference is that you have factor `self._norm_factor` in one case and not in the other case, so I think it is a reasonable name (or `self._loss_value * self._norm_factor` would be wrong then, right?).

Reply:
I am just saying that my initial code had a slightly different behaviour. But yes, I can add it as always normalized and un-normalized. I will also add `debug_grad_summaries` to the docs.